File: _CodeTopics/LeetCode_contest/biweekly/biweekly2021/48/48_1.py
class Solution(object):
    def secondHighest(self, s):
        """
        :type s: str
        :rtype: int
        """
        
        stk = [-1] * 10
        for ch in s:
            if ch.isdigit():
                stk[int(ch)] = int(ch)
        
        flag = 0
        for i in range(9, -1, -1):
            if stk[i] != -1:
                if flag == 0:
                    flag = 1
                elif flag == 1:
                    return i
        return -1
    
"""
https://leetcode-cn.com/submissions/detail/157679832/

300 / 300 个通过测试用例
状态：通过
执行用时: 40 ms
内存消耗: 13 MB
"""

File: _CodeTopics/LeetCode/1601-1800/001796--biweekly48_1/tran/48_1.cpp
class Solution {
public:
    int secondHighest(string s) {

        // 第 48 场双周赛第一题
        // https://leetcode.cn/submissions/detail/157679832/

        vector<int> stk(10, 0);
        for (char ch : s) {
            if (isdigit(ch)) {
                // 也可以用： stk[ch-48] = 1;
                stk[ch-'0'] = 1;
            }
        }

        bool flag = false;
        for (int i = 9; i > -1; --i) {
            if (stk[i] > 0) {
                if (!flag) {
                    flag = true;
                } else {
                    return i;
                }
            }
        }
        return -1;
    }
};

/*
https://leetcode.cn/submissions/detail/386802962/

执行用时：
4 ms
, 在所有 C++ 提交中击败了
65.94%
的用户
内存消耗：
6.6 MB
, 在所有 C++ 提交中击败了
62.86%
的用户
通过测试用例：
301 / 301
*/
